data/utils.sh

```diff
@@ -0,0 +1,213 @@
+    done
+}
+
+# all the disabled services + the oneshot daemons
+check_inactive_services()
+{
+    for svc in $1; do
+        svcStatus="$(snap services edgexfoundry.$svc | grep $svc | awk '{print $3}')"
+        if [ "inactive" != "$svcStatus" ]; then
+            echo "service $svc has status \"$svcStatus\" but should be inactive"
+            if [ "$FATAL" = "1" ]; then
+                exit 1
+            fi
+        fi
+    done
+}
+
+get_snap_svc_status()
+{
+    case "$1" in 
+        "status")
+            svcStatus="$(snap services edgexfoundry.$1 | grep $1 | awk '{print $3}')"
+            ;;
+        "")
+            ;;
+    esac
+}
+
+snap_remove()
+{
+    snap remove edgexfoundry || true
+}
+
+# wait for services to come online
+# NOTE: this may have to be significantly increased on arm64 or low RAM platforms
+# to accomodate time for everything to come online
+snap_wait_all_services_online()
+{
+    http_error_code_regex="4[0-9][0-9]|5[0-9][0-9]"
+    all_services_online=false
+    i=0
+
+    while [ "$all_services_online" = "false" ];
+    do
+        echo "waiting for all services to come online. Current retry count: $i/300"
+        #max retry avoids forever waiting
+        ((i=i+1))
+        if [ "$i" -ge 300 ]; then
+            echo "services timed out, reached max retry count of 300"
+            exit 1
+        fi
+
+        #dial services
+        core_data_status_code=$(curl --insecure --silent --include \
+            --connect-timeout 2 --max-time 5 \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59880/api/v2/ping') || true
+        core_metadata_status_code=$(curl --insecure --silent --include \
+            --connect-timeout 2 --max-time 5 \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59881/api/v2/ping') || true
+        core_command_status_code=$(curl --insecure --silent --include \
+            --connect-timeout 2 --max-time 5 \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59882/api/v2/ping') || true
+
+        #error status 4xx/5xx will fail the test immediately
+        if [[ $core_data_status_code =~ $http_error_code_regex ]] \
+            || [[ $core_metadata_status_code =~ $http_error_code_regex ]] \
+            || [[ $core_command_status_code =~ $http_error_code_regex ]]; then
+            echo "core service(s) received status code 4xx or 5xx"
+            exit 1
+        fi
+
+        if [[ "$core_data_status_code" == 200 ]] \
+            && [[ "$core_metadata_status_code" == 200 ]] \
+            && [[ "$core_command_status_code" == 200 ]] \
+            && [ -n "$(lsof -i -P -n -S 2 | grep 8000)" ] \
+            && [ -n "$(lsof -i -P -n -S 2 | grep 8200)" ] \
+            && [ -n "$(lsof -i -P -n -S 2 | grep 8500)" ] \
+            && [ -n "$(lsof -i -P -n -S 2 | grep 5432)" ] \
+            && [ -n "$(lsof -i -P -n -S 2 | grep 6379)" ]; then
```

Comment:
Why not use the `snap_wait_port_status` function for these?

Reply:
Updated. Thanks!